File: java/cdk8s-crd/src/main/java/imports/io/cert_manager/CertificateSpecPrivateKeyRotationPolicy.java
package imports.io.cert_manager;

/**
 * RotationPolicy controls how private keys should be regenerated when a re-issuance is being processed.
 * <p>
 * If set to Never, a private key will only be generated if one does not already exist in the target <code>spec.secretName</code>. If one does exists but it does not have the correct algorithm or size, a warning will be raised to await user intervention. If set to Always, a private key matching the specified requirements will be generated whenever a re-issuance occurs. Default is 'Never' for backward compatibility.
 * <p>
 * Default: Never' for backward compatibility.
 */
@javax.annotation.Generated(value = "jsii-pacmak/1.79.0 (build b22f628)", date = "2023-03-31T14:16:45.751Z")
@software.amazon.jsii.Jsii(module = imports.io.cert_manager.$Module.class, fqn = "iocert-manager.CertificateSpecPrivateKeyRotationPolicy")
public enum CertificateSpecPrivateKeyRotationPolicy {
    /**
     * Never.
     */
    NEVER,
    /**
     * Always.
     */
    ALWAYS,
}

File: go/cdk8s-crd/imports/certmanagerio/CertificateSpecPrivateKeyRotationPolicy.go
package certmanagerio


// RotationPolicy controls how private keys should be regenerated when a re-issuance is being processed.
//
// If set to `Never`, a private key will only be generated if one does not
// already exist in the target `spec.secretName`. If one does exist but it
// does not have the correct algorithm or size, a warning will be raised
// to await user intervention.
// If set to `Always`, a private key matching the specified requirements
// will be generated whenever a re-issuance occurs.
// Default is `Always`.
// The default was changed from `Never` to `Always` in cert-manager >=v1.18.0.
// The new default can be disabled by setting the
// `--feature-gates=DefaultPrivateKeyRotationPolicyAlways=false` option on
// the controller component.
// Default: Always`.
//
type CertificateSpecPrivateKeyRotationPolicy string

const (
	// Never.
	CertificateSpecPrivateKeyRotationPolicy_NEVER CertificateSpecPrivateKeyRotationPolicy = "NEVER"
	// Always.
	CertificateSpecPrivateKeyRotationPolicy_ALWAYS CertificateSpecPrivateKeyRotationPolicy = "ALWAYS"
)

